### PR TITLE
ideal flow for DevOps

### PR DESCRIPTION
**Pull request by samhita-alla**
Signed-off-by: Samhita Alla <aallasamhita@gmail.com>

I used DevOps but not MLOps because MLOps may mislead users into thinking this is only for ML pipelines.

Fixes https://github.com/flyteorg/flyte/issues/1985

### CODE REVIEW

File: rsts/deployment/devops.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
+.. _devops-ideal-flow:
+
+Ideal Flow for DevOps
```
Comment (kumare3) [inline rsts/deployment/devops.rst:3]: I think this should not say devops, we can just say how to maintain a good workflow?

File: rsts/deployment/devops.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+---------------------
+
+Flyte could be useful for many use cases, be it model training, data processing, ELT/ETL, or bioinformatics.
+When workflows are built and deployed, be it in any domain, we may need automation to reduce human-in-the-loop to some extent;
+essentially, to automate serializing and registering workflows, creating docker containers, etc.
+
+Before diving into an example use case that explains how DevOps could power Flyte pipelines, let's look at some DevOps-style features Flyte ships with.
+
+- Every DAG is versioned with Git commit's SHA, which is modifiable
```
Comment (kumare3) [inline rsts/deployment/devops.rst:12]: What does modifiable mean?

File: rsts/deployment/devops.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+- Tasks and workflows are immutable
```
Comment (kumare3) [inline rsts/deployment/devops.rst:13]: A version is immutable, any mutation results in a new version

File: rsts/deployment/devops.rst [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+- In the case of rapid iteration of code with no dependency modification, :ref:`deployment-fast-registration` can be used
+- Multi-tenancy support through projects and domains
+- Executions can be monitored using logs on :ref:`Flyte UI <ui>`
+
+Lyft's Use Case
```
Comment (kumare3) [inline rsts/deployment/devops.rst:18]: Case study - MLops at lyft?

File: rsts/deployment/devops.rst [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+Flyte powers more than 1 million model training and data processing executions per month at Lyft.
+DevOps automation has been made part of Flyte pipelines to handle such a massive number of executions.
+
+- Every time a new PR is created:
+
+  - A Docker container is built from the PR
+  - Tasks and workflows are registered with Flyte
+- If dependencies are modified, PR securely builds the container again
+- For just code changes, fast registration is used
```
Comment (kumare3) [inline rsts/deployment/devops.rst:29]: Can be done from laptop, users are not permitted to push docker containers from local machines

File: rsts/deployment/devops.rst [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+- Once a PR is merged, the user simulates a deployment through a deployment pipeline
+- At each stage of the deployment, tasks and workflows are registered with a specific domain in Flyte,
+  where each domain may change the data directory, associated roles, or some meta attributes like labels and annotations
+- For production deployments, logs and metrics are automatically tracked, and models are promoted to serving infrastructure
+- Users can use interactive notebooks to retrieve intermediate or final outputs, analyze data and automate various monitoring tasks
+
+Thus, everything is automatically tracked. In fact, multiple users can create isolated PRs and test independently.
+
+Implementing DevOps when running Flyte pipelines is essential when Flyte is deployed to production.
```
Comment (kumare3) [inline rsts/deployment/devops.rst:38]: Is not essential

File: rsts/deployment/devops.rst (hunk 1 above)
Comment (samhita-alla) [reply rsts/deployment/devops.rst]: Now using the title: "How to Streamline Your Flyte Workflows"

**Review by kumare3 — APPROVED** [review]

**Pull request merged**